server/memphis_handlers_tags.go

```diff
@@ -151,6 +150,91 @@ func AddTagsToEntity(tags []models.CreateTag, entity_type string, entity_id prim
 	return nil
 }
 
+func CreateNewTag(tagToCreate models.CreateTag) (error, models.Tag) {
+	name := strings.ToLower(tagToCreate.Name)
+	exist, _, err := IsTagExist(name)
+	if err != nil {
+		return err, models.Tag{}
+	}
+	var newTag models.Tag
+	if !exist {
+		stationArr := []primitive.ObjectID{}
+		schemaArr := []primitive.ObjectID{}
+		userArr := []primitive.ObjectID{}
+		newTag = models.Tag{
+			ID:       primitive.NewObjectID(),
+			Name:     name,
+			Color:    tagToCreate.Color,
+			Stations: stationArr,
+			Schemas:  schemaArr,
+			Users:    userArr,
+		}
+
+		filter := bson.M{"name": newTag.Name}
+		update := bson.M{
+			"$setOnInsert": bson.M{
+				"_id":      newTag.ID,
+				"name":     newTag.Name,
+				"color":    newTag.Color,
+				"stations": newTag.Stations,
+				"schemas":  newTag.Schemas,
+				"users":    newTag.Users,
+			},
+		}
+		opts := options.Update().SetUpsert(true)
+		_, err = tagsCollection.UpdateOne(context.TODO(), filter, update, opts)
+		if err != nil {
+			return err, models.Tag{}
+		}
+
+	}
+	if err != nil {
+		return err, models.Tag{}
+	}
+
+	return nil, newTag
+}
+
+func AddTagToEntity(color string, name string, entity_type string, entity_id primitive.ObjectID) error {
+	switch entity_type {
+	case "station":
+		tag_name := strings.ToLower(name)
+		exist, tag, err := IsTagExist(tag_name)
+		if err != nil {
+			return err
+		}
+		if !exist {
+			err = CreateTag(tag_name, entity_type, entity_id, color)
```

Comment:
you are checking whether the tag exists and then again inside this function checking it again, why?